xarray/tests/test_plot.py

```diff
@@ -267,6 +268,35 @@ def test2d_1d_2d_coordinates_contourf(self):
         a.plot.contourf(x="time", y="depth")
         a.plot.contourf(x="depth", y="time")
 
+    def test_contourf_cmap_set(self):
+        sz = (4, 4)
+        a = DataArray(easy_array(sz), dims=["z", "time"])
+
+        cmap = copy(mpl.cm.viridis)
+
+        cmap.set_bad("w")
+        # check we actually changed the set_bad color
+        assert cmap._rgba_bad != mpl.cm.viridis._rgba_bad
+
+        cmap.set_under("w")
+        # check we actually changed the set_under color
+        assert cmap._rgba_under != mpl.cm.viridis._rgba_under
+
+        cmap.set_over("w")
+        # check we actually changed the set_over color
+        assert cmap._rgba_over != mpl.cm.viridis._rgba_over
+
+        pl = a.plot.contourf(cmap=copy(cmap))
```

Comment:
Why do you apply `copy` here? I think it is unnecessary.

Reply:
The idea is that since we test the output against `cmap` we want to be totally sure `cmap` isn't changed by `a.plot.contourf()` (even though it should not be changed anyway). Actually, I think we should `deepcopy` instead of `copy` to achieve this.